reporting/src/main/java/org/wso2/testgrid/reporting/TestReportEngine.java

```diff
@@ -825,6 +825,7 @@ private void postProcessSummaryTable(Map<String, InfrastructureBuildStatus> test
                 Optional<TestPlan> testPlanById = testPlanUOW.getTestPlanById(testPlanYaml.getId());
                 if (testPlanById.isPresent()) {
                     TestPlan mergedTestPlan = TestGridUtil.mergeTestPlans(testPlanYaml, testPlanById.get(), false);
+                    mergedTestPlan.setWorkspace(workspace);
```

Comment:
Setting workspace like this in a unrelated class like TestReportEngine really complicate the matters in the long run! Shall we see a better way to fix this?

Reply:
When we create a TestPlan object in the middle of the flow by calling the database or so, it is missing the workspace property. So in such times the rule would be; **whenever a TestPlan object is created, right then we need to set its workspace (which we receive at runtime via input argument)**. 
Agreed it complicates the process :) 
The ideal solution would be to keep the `workspace` in a Context where we keep run-time details of the Job. :)